iot_config/src/route.rs

```diff
@@ -180,19 +183,27 @@ pub async fn update_route(
 async fn insert_euis(
     euis: &[EuiPair],
     db: impl sqlx::PgExecutor<'_>,
-) -> Result<(), RouteStorageError> {
-    // We don't want to take any actions if a route_id cannot be parsed.
-    let mut eui_values = vec![];
-    for eui in euis.iter() {
-        eui_values.push((
-            Uuid::try_parse(&eui.route_id)?,
-            i64::from(eui.app_eui),
-            i64::from(eui.dev_eui),
-        ));
+) -> Result<Vec<EuiPair>, RouteStorageError> {
+    if euis.is_empty() {
+        return Ok(vec![]);
     }
+    // We don't want to take any actions if a route_id cannot be parsed.
+    let eui_values = euis
+        .iter()
+        .map(|eui_pair| {
+            Uuid::try_parse(&eui_pair.route_id).map(|uuid| {
```

Comment:
does implementing `TryFrom` for `&EuiPair` make sense here?

Reply:
when i first thought about that i imagined putting it in `src/lora_field.rs` but that would break the encapsulation of the route ID since the only thing that actually cares that it's a Uuid is sqlx but then i figured i could just put the `From` impl in this module
and also it's pretty academic since this is not a library and the very idea of `From` sort of breaks encapsulation by intention...